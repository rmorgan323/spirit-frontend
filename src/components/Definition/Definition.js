/*eslint-disable no-useless-escape*/

import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import './Definition.css';

const Definition = props => {
  let relevantDefinitions;
  let display;
  const firstWord = new RegExp(/^([\w\-]+)/);

  if (props.definitions.length) {
    relevantDefinitions = props.definitions.filter(def => {
      return (
        def.term.match(firstWord)[0].toLowerCase() ===
        props.title.match(firstWord)[0].toLowerCase()
      );
    });

    if (!relevantDefinitions.length) {
      relevantDefinitions = [{ term: 'No Definitions Found', definition: '' }];
    }

    display = relevantDefinitions.map((def, index) => {
      return (
        <div key={index}>
          <h5>{def.term}</h5>
          <p>{def.definition}</p>
          <hr />
        </div>
      );
    });
  }

  return <div className="Definition">{display}</div>;
};

const mapStateToProps = store => ({
  definitions: store.definitions
});

export default connect(mapStateToProps, null)(Definition);

Definition.propTypes = {
  definitions: PropTypes.object,
  title: PropTypes.string
};
