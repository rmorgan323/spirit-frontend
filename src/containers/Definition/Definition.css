/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.Definition {
  background-color: rgba(150, 150, 150, 0.9);
  position: absolute;
  top: 26px;
  left: 0;
  z-index: 10;
  width: 300px;
  padding: 10px; }
  .Definition .definition-title {
    color: #fff;
    margin: 0;
    font-size: 18px; }
  .Definition .definition-body {
    color: #fff;
    text-align: left;
    margin: 10px 0 0;
    font-size: 14px;
    font-weight: 100; }
  .Definition hr {
    border: 1px solid rgba(0, 0, 0, 0.2); }
