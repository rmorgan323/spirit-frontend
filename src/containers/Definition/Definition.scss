@import '../../variables.scss';

.Definition {
  background-color: rgba(150,150,150, .9);
  position: absolute;
  top: 26px;
  left: 0;
  z-index: 10;
  width: 300px;
  padding: 10px;

  .definition-title {
    color: #fff;
    margin: 0;
    font-size: 18px;
  }

  .definition-body {
    color: #fff;
    text-align: left;
    margin: 10px 0 0;
    font-size: 14px;
    font-weight: 100;
  }

  hr {
    border: 1px solid rgba(0,0,0,.2)
  }
 }