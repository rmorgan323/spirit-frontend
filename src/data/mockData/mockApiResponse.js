const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;