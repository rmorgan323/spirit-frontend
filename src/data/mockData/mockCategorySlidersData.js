/*eslint-disable camelcase*/

export const mockTitle = 'Capacities';

export const mockSlidersArray = [
  'Attention',
  'Action',
  'Affect',
  'Autonomic',
  'Arousability'
];

export const mockDatabaseNamesArray = [
  'mod_2_attention',
  'mod_2_action',
  'mod_2_affect',
  'mod_2_autonomic',
  'mod_2_arousability'
];

export const mockReminderAsterisk = [null, '*', '*', '*', '*'];

export const mockReminder = '* If selected, remember to make a hypothesized arousal chart.';

export const mockSliderTitle = 'Attention';

export const mockDatabaseName = 'mod_2_attention';

