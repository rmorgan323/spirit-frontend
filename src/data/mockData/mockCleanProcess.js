/*eslint-disable camelcase*/

export const mockCleanProcess = {
  id: 10,
  sen_h_vestibular: null,
  sen_h_proprioception: null,
  sen_h_tactile: null,
  sen_h_auditory: null,
  sen_h_visual: null,
  sen_h_intero: null,
  sen_1_ability: null,
  sen_1_match: null,
  sen_1_classification: null,
  sen_1_spatial: null,
  sen_1_figure: null,
  sen_1_inferences: null,
  sen_2_initiation: null,
  sen_2_imitation: null,
  sen_2_affordance: null,
  sen_2_generativity: null,
  sen_3_sequencing: null,
  sen_3_sequencing_gestalt: null,
  sen_3_schema_selection: null,
  sen_3_schema_expansion: null,
  sen_3_modification: null,
  sen_4_timing: null,
  sen_4_timing_force: null,
  sen_4_spatial: null,
  sen_4_temporal: null,
  sen_4_control: null,
  sen_4_monitoring: null,
  sen_5_oral: null,
  sen_5_ocular: null,
  sen_5_postural: null,
  sen_5_limb: null,
  sen_5_gross: null,
  sen_5_fine: null,
  sen_5_tool: null,
  sen_5_projected: null,
  sen_6_somatodyspraxia: null,
  sen_6_visual: null,
  sen_6_generalized: null,
  sen_6_bmc: null,
  sen_6_praxis: null,
  mod_h_vestibular: null,
  mod_h_proprioception: null,
  mod_h_tactile: null,
  mod_h_auditory: null,
  mod_h_visual: null,
  mod_h_intero: null,
  mod_1_defensiveness: null,
  mod_1_over: null,
  mod_1_under: null,
  mod_1_variable: null,
  mod_2_attention: null,
  mod_2_action: null,
  mod_2_affect: null,
  mod_2_autonomic: null,
  mod_3_physiological: null,
  mod_3_sensory: null,
  mod_3_motor: null,
  mod_3_emotional: null,
  mod_3_cognitive: null,
  mod_3_social: null,
  mod_3_prosocial: null,
  mod_4a_passive: null,
  mod_4a_co_regulation: null,
  mod_4a_approach: null,
  mod_4a_adaptive: null,
  mod_4b_active: null,
  mod_4b_self_regulation: null,
  mod_4b_avoid: null,
  mod_4b_maladaptive: null,
  exe_1_orient: null,
  exe_1_inhibition: null,
  exe_2a_working: null,
  exe_2a_shifting: null,
  exe_2b_verbal: null,
  exe_2b_spatial: null,
  exe_2b_general: null,
  exe_2b_global: null,
  exe_2b_flexibility: null,
  exe_3_sequencing: null,
  exe_3_monitoring: null,
  exe_4a_future: null,
  exe_4a_spatial: null,
  exe_4a_problem: null,
  exe_4a_reflective: null,
  exe_4a_internalized: null,
  exe_4b_motivational: null,
  exe_4b_effortful: null,
  exe_4b_self_control: null,
  exe_4b_self_regulation: null,
  exe_4b_self_compassion: null,
  pos_h_vestibular: null,
  pos_h_proprioception: null,
  pos_h_tactile: null,
  pos_h_auditory: null,
  pos_h_visual: null,
  pos_h_intero: null,
  pos_2_stability: null,
  pos_2_mobility: null,
  pos_2_tonic: null,
  pos_2_phasic: null,
  pos_2_muscle: null,
  pos_3_sagittal: null,
  pos_3_frontal: null,
  pos_3_transverse: null,
  pos_3_midlines: null,
  pos_5_suck: null,
  pos_5_ocular: null,
  pos_5_antigravity: null,
  pos_5_reflex: null,
  pos_5_alignment_COG: null,
  pos_5_alignment_orient: null,
  pos_5_anticipatory: null,
  pos_5_interactive: null,
  pos_6_push: null,
  pos_6_bilateral: null,
  pos_6_grasp: null,
  pos_6_strength: null,
  pos_6_effortful: null,
  soc_1_regulation: null,
  soc_1_engagement: null,
  soc_1_reciprocity: null,
  soc_1_problem: null,
  soc_1_representation: null,
  soc_1_triangular: null,
  soc_1_comparative: null,
  soc_1_reflective: null,
  soc_2_self_awareness: null,
  soc_2_joint: null,
  soc_2_social_referencing: null,
  soc_2_imitation: null,
  soc_2_inferential: null,
  soc_2_contingent: null,
  soc_2_theory: null,
  soc_2_central: null,
  soc_2_mentalizing: null,
  soc_2_empathy: null,
  soc_2_social_motivators: null,
  soc_2_pro_social: null,
  soc_3_affective: null,
  soc_3_emotional_contagion: null,
  soc_3_activation_low: null,
  soc_3_emotional_cascade: null,
  soc_3_activation_high: null,
  soc_3_emotional_differentiation: null,
  soc_3_emotional_association: null,
  soc_3_self_control: null,
  soc_3_self_regulation: null,
  soc_3_self_compassion: null,
  soc_4_mental: null,
  soc_4_thoughts: null,
  soc_4_attachments: null,
  soc_4_trauma: null,
  session_id: 10,
  created_at: "2018-02-20T21:25:45.454Z",
  updated_at: "2018-02-20T21:25:45.454Z",
  mod_2_arousability: null,
  exe_3_initiation: null,
  exe_3_planning: null,
  pos_4_core: null,
  pos_4_shoulder: null,
  pos_4_pelvic: null,
  pos_4_head: null,
  pos_4_eyes: null,
  pos_4_hand: null,
  pos_4_lower: null,
  pos_4_foot: null,
  soc_2_social_perspective: null,
  pos_1_outer: null,
  pos_1_inner: null,
  exe_4a_generativity: null
};
