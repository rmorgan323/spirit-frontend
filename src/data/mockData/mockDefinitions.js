/*eslint-disable camelcase*/
/*eslint-disable max-len*/

export const mockDefinitions = [
  {
    term: 'JRC/AR',
    definition: `This is the abbreviation for Just Right Challenge and Adaptive Response. Since we are utilizing and ASI© model as one foundational frame of reference, the treatment plan needs to include mindful decisions about how the therapists might guide the JRC and what AR's are noted.  This is where you should note what treatment was offered and what the response was.  In the treatment planning section, do this related to the STEPSI© components.`,
    category_name: 'Spirit',
    imageURL: ''
  },
  {
    term: 'S-A-M',
    definition: `This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the "driver" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.`,
    category_name: 'Spirit',
    imageURL: 'true'
  },
  {
    term: 'S in S-A-M: Sensory',
    definition: `High: Sensory Discrimination, Low: Sensory Modulation`,
    category_name: 'Spirit',
    imageURL: ''
  },
  {
    term: 'A in S-A-M: Affective',
    definition: `High: Motivational Biasing System, Low: Affective Valence`,
    category_name: 'Spirit',
    imageURL: ''
  },
  {
    term: 'M in S-A-M: Motor',
    definition: `High: Motor Planning Executive system, Low: Automatic postural and motor functions`,
    category_name: 'Spirit',
    imageURL: ''
  }
];
