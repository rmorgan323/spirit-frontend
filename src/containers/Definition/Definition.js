/*eslint-disable no-useless-escape*/

import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import './Definition.css';

export const Definition = props => {
  let relevantDefinitions = [];
  let display;
  const firstWord = new RegExp(/^([\w\-]+)/);

  if (props.definitions.length) {
    props.definitions.forEach(def => {
      if (def.term.toLowerCase() === props.title.toLowerCase()) {
        relevantDefinitions.push(def);
      }
    });

    if (!relevantDefinitions.length) {
      relevantDefinitions = props.definitions.filter(def => {
        return (
          def.term.match(firstWord)[0].toLowerCase() ===
          props.title.match(firstWord)[0].toLowerCase()
        );
      });
    }

    if (!relevantDefinitions.length) {
      relevantDefinitions = [{ term: 'No Definitions Found', definition: '' }];
    }

    display = relevantDefinitions.map((def, index) => {
      return (
        <div key={index}>
          <h5 className="definition-title">{def.term}</h5>
          <p className="definition-body">{def.definition}</p>
          <hr />
        </div>
      );
    });
  }

  return <div className="Definition">{display}</div>;
};

export const mapStateToProps = store => ({
  definitions: store.definitions
});

export default connect(mapStateToProps, null)(Definition);

Definition.propTypes = {
  definitions: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  title: PropTypes.string
};
