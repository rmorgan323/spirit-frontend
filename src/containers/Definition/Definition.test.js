import React from 'react';
import { Definition, mapStateToProps } from './Definition';
import { shallow } from 'enzyme';
import { mockDefinitions } from '../../data/mockData/mockDefinitions';

describe('Definition tests', () => {
  let definitions;
  let renderedDefinition;
  let title;

  beforeEach(() => {
    definitions = mockDefinitions;
    title = 'S-A-M';

    renderedDefinition = shallow(
      <Definition definitions={definitions} title={title} />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedDefinition).toMatchSnapshot();
  });
});

describe('mapStateToProps tests', () => {
  it('Should pull definitions from store', () => {
    const definitions = mockDefinitions;
    const mockStore = { definitions };
    const result = mapStateToProps(mockStore);

    expect(result.definitions).toEqual(mockStore.definitions);
  });
});
