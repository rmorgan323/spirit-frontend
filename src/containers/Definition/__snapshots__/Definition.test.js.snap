// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    definitions={
      Array [
        Object {
          "category_name": "Spirit",
          "definition": "This is the abbreviation for Just Right Challenge and Adaptive Response. Since we are utilizing and ASI© model as one foundational frame of reference, the treatment plan needs to include mindful decisions about how the therapists might guide the JRC and what AR's are noted.  This is where you should note what treatment was offered and what the response was.  In the treatment planning section, do this related to the STEPSI© components.",
          "imageURL": "",
          "term": "JRC/AR",
        },
        Object {
          "category_name": "Spirit",
          "definition": "This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the \\"driver\\" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.",
          "imageURL": "true",
          "term": "S-A-M",
        },
        Object {
          "category_name": "Spirit",
          "definition": "High: Sensory Discrimination, Low: Sensory Modulation",
          "imageURL": "",
          "term": "S in S-A-M: Sensory",
        },
        Object {
          "category_name": "Spirit",
          "definition": "High: Motivational Biasing System, Low: Affective Valence",
          "imageURL": "",
          "term": "A in S-A-M: Affective",
        },
        Object {
          "category_name": "Spirit",
          "definition": "High: Motor Planning Executive system, Low: Automatic postural and motor functions",
          "imageURL": "",
          "term": "M in S-A-M: Motor",
        },
      ]
    }
    title="S-A-M"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h5
            className="definition-title"
          >
            S-A-M
          </h5>
          <p
            className="definition-body"
          >
            This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the "driver" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.
          </p>
          <hr />
        </div>,
      ],
      "className": "Definition",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5
              className="definition-title"
            >
              S-A-M
            </h5>,
            <p
              className="definition-body"
            >
              This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the "driver" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.
            </p>,
            <hr />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "S-A-M",
              "className": "definition-title",
            },
            "ref": null,
            "rendered": "S-A-M",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the \\"driver\\" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.",
              "className": "definition-body",
            },
            "ref": null,
            "rendered": "This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the \\"driver\\" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h5
              className="definition-title"
            >
              S-A-M
            </h5>
            <p
              className="definition-body"
            >
              This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the "driver" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.
            </p>
            <hr />
          </div>,
        ],
        "className": "Definition",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="definition-title"
              >
                S-A-M
              </h5>,
              <p
                className="definition-body"
              >
                This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the "driver" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.
              </p>,
              <hr />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "S-A-M",
                "className": "definition-title",
              },
              "ref": null,
              "rendered": "S-A-M",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the \\"driver\\" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.",
                "className": "definition-body",
              },
              "ref": null,
              "rendered": "This is an abbreviation for Sensory-Affective-Motor, and was originated by Greenspan. For the SpIRiT©, Stackhouse and colleagues have elaborated the Sensory-Affective-Motor function into the low and high routes of processing that occur in each function.  This is the pivotal function that helps to 'connect the dots and trace the threads' of connection between the functions in the SpIRiT© model.  The idea here is for the clinician to select which aspect seems to be the \\"driver\\" of the concerns in that area (S-A-M) - it is a clinical decision/tacit response about what seems to be the most primary. This should be consistent, then, with what your clinical decisions are in each area.  It could be one, two, or all three (S-A-M), and it could vary from one area to another.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
